### DISCUSSION TITLE
How do I link the phase and phase domain on Multiphase Flow?

### DISCUSSION THREAD

**Discussion by MayconSilva7** [ansys/pyfluent#1946] [General]
Hello, sorry for sending again a message. It is because I have tried so many times, but I did not get any of them.

On Gentop multiphase model, when I named the 3 phases, I don't get to set link the to phase domain. The definition in Pythons says:

`` `` `` 
def phase_domains(self, *args, **kwargs):
                    """
                    Enters the menu to select a specific phase.
                    """
                    return PyMenu(self.service, "/define/phases/set_domain_properties/phase_domains").execute(*args, **kwargs)
`` `` ``
I have tried the following commands:

`` `` `` 
solver.tui.define.phases.set_domain_properties.phase_domains("0","water-liquid")
`` `` ``

Command 2: 

`` `` `` 
solver.tui.define.phases.set_domain_properties.phase_domains("Primary Phase","water-liquid")
`` `` ``

Command 3: 
`` `` `` 
solver.tui.define.phases.set_domain_properties.phase_domains("continuous-liquid","water-liquid")
`` `` ``

I set three phases:
>> continuous liquid
>> dispersed gas 
>> hybrid-gentop-gas

Thank you very much.

**Comment by MayconSilva7** [comment]
Please, can anyone help me with this?

Thank you very much!

**Reply by mkundu1** [reply]
Do you know the corresponding TUI command in Fluent? You can then transform it to the Python command using [transformation rule](https://fluent.docs.pyansys.com/version/stable/user_guide/tui_commands.html#tui-command-transformation-rules).

**Reply by MayconSilva7** [reply]
Hello, I am sorry for the delay. Thank you very much for your helping and this tip. In fact, I got resolve this using the Fluent console and then converting to Python language (TUI commands). 

But I would like to ask for help in another thing, because I have been trying set the population balance model through the TUI commands as follows:

`` `` ``
tui.define.models.multiphase.population_balance.model='inhomo_discrete'
`` `` ``

I have tried another combination as follows:

`` `` ``
tui.define.models.multiphase.population_balance.model=("inhomo_discrete", "2","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes")
`` `` ``

However, I did not get to set the model. Please help me, and thank you very much in advance.

**Reply by mkundu1** [reply]
Through `tui`, we can only execute functions with positional arguments. So, the command would be 
```
tui.define.models.multiphase.population_balance.model("inhomo_discrete","2","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes")
```
assuming these are the positional arguments.

**Reply by MayconSilva7** [reply]
Hello, Thank you very much, 

I have tried, but I received the following message: "Invalid choice". I did not get any option.

Thank you very much in advance.

**Reply by MayconSilva7** [reply]
I have tried this following code as well, but I did not get it:

`` `` ``
solver.tui.define.models.multiphase.population_balance.model="inhomo_discrete"
`` `` ``


Kind regards,

**Reply by MayconSilva7** [reply]
Please, @mkundu1, Can you help with this? 

I have been trying to set up the GENTOP method from the beginning. When I try to set up the population balance model, I din't get it.

**Reply by raph-luc** [reply]
> I have tried this following code as well, but I did not get it:
> 
> ` ` `solver.tui.define.models.multiphase.population_balance.model="inhomo_discrete"` ` `
> 
> Kind regards,

@MayconSilva7 That is not a valid PyFluent TUI command. A valid TUI command equivalent to what you are trying there would look like `solver.tui.define.models.multiphase.population_balance.model("inhomo_discrete")`.

Make sure to read and understand this: 
https://fluent.docs.pyansys.com/version/dev/user_guide/tui_commands.html

After reading that, I would suggest you first figure out, inside a standalone Fluent session (not using PyFluent), what is the exact sequence of TUI commands required to achieve what you need, and then transform it to the equivalent PyFluent TUI command as described in the link above.

If you don't know how to use TUI commands inside Fluent take a look at the Fluent documentation.
